internal/repositories/github/webhooks/manager.go

```diff
@@ -0,0 +1,179 @@
+		client GitHubWebhookClient,
+		repoOwner string,
+		repoName string,
+	) (string, *github.Hook, error)
+
+	DeleteWebhook(
+		ctx context.Context,
+		client GitHubWebhookClient,
+		repoOwner string,
+		repoName string,
+		hookID int64,
+	) error
+}
+
+type webhookManager struct {
+	webhookConfig server.WebhookConfig
+}
+
+// NewWebhookManager instantiates an instances of the WebhookManager interface
+func NewWebhookManager(webhookConfig server.WebhookConfig) WebhookManager {
+	return &webhookManager{webhookConfig: webhookConfig}
+}
+
+var (
+	targetedEvents = []string{"*"}
+)
+
+// CreateWebhook creates a Minder-managed webhook in the specified GitHub repo
+// Note that this method will delete any previous Minder-managed webhooks
+// before attempting to make a new one.
+// Returns the UUID of this webhook, along with the API response from GitHub
+// with details of the new webhook
+func (w *webhookManager) CreateWebhook(
+	ctx context.Context,
+	client GitHubWebhookClient,
+	repoOwner string,
+	repoName string,
+) (string, *github.Hook, error) {
+	// generate unique URL for this webhook
+	baseURL := w.webhookConfig.ExternalWebhookURL
+	hookUUID := uuid.New().String()
+	webhookURL := fmt.Sprintf("%s/%s", baseURL, hookUUID)
+	parsedWebhookURL, err := url.Parse(webhookURL)
+	if err != nil {
+		return "", nil, err
+	}
+
+	// If we have an existing hook for same repo, delete it
+	if err := w.cleanupStaleHooks(ctx, client, repoOwner, repoName, parsedWebhookURL.Host); err != nil {
+		return "", nil, err
+	}
+
+	// Attempt to register new webhook
+	ping := w.webhookConfig.ExternalPingURL
+	secret := w.webhookConfig.WebhookSecret
+	newHook := &github.Hook{
+		Config: map[string]any{
+			"url":          webhookURL,
+			"content_type": "json",
+			"ping_url":     ping,
+			"secret":       secret,
+		},
+		Events: targetedEvents,
+	}
+
+	webhook, err := client.CreateHook(ctx, repoOwner, repoName, newHook)
+	if err != nil {
+		return "", nil, fmt.Errorf("error creating hook: %w", err)
+	}
+
+	return hookUUID, webhook, nil
+}
+
+// DeleteWebhook deletes the specified webhook from the specified GitHub repo
+// Note that deletions of non-existent webhooks are treated as no-ops
+func (_ *webhookManager) DeleteWebhook(
+	ctx context.Context,
+	client GitHubWebhookClient,
+	repoOwner string,
+	repoName string,
+	hookID int64,
+) error {
+	resp, err := client.DeleteHook(ctx, repoOwner, repoName, hookID)
+	if err != nil {
+		if resp != nil && resp.StatusCode == http.StatusNotFound {
+			// if the hook is not found, we can ignore the error, user might have deleted it manually
+			return nil
+		}
+		return fmt.Errorf("error deleting hook: %w", err)
+	}
+
+	return nil
+}
+
+func (w *webhookManager) cleanupStaleHooks(
+	ctx context.Context,
+	client GitHubWebhookClient,
+	repoOwner string,
+	repoName string,
+	webhookHost string,
+) error {
+	logger := zerolog.Ctx(ctx).With().
+		Str("repoName", repoName).
+		Str("repoOwner", repoOwner).Logger()
```

Comment:
If you do this out in `registerWebhookForRepository` and store it back in `ctx`, you can simply use `logger := zerolog.Ctx(ctx)` here.